lib/memoize.js

```diff
@@ -1,86 +1,144 @@
-function isPrimitive(value) {
```

Comment:
It is still there, see line 43. It seems that it pretty much inlines your original code using simple objects instead of classes and reduces new memory allocation and inner function calls